### PR TITLE
branch-off: make release numbers more consistent

### PR DESCRIPTION
**Pull request by figsoda**

### CODE REVIEW

File: src/Branch-Off.md [hunk 1]
```diff
@@ -12,18 +12,18 @@ infrastructure team and the release team to work more asynchronously. Reach out
...
-    - release-22.11-small
+    - release-23.05
+    - release-23.05-small
 - [https://hydra.nixos.org/project/nixpkgs](https://hydra.nixos.org/project/nixpkgs)
-    - nixpkgs-22.11-darwin
-    - staging-next-22.11
+    - nixpkgs-23.05-darwin
+    - staging-next-23.05
 
 Example configuration: [nixos:release-22.11](https://hydra.nixos.org/jobset/nixos/release-22.11#tabs-configuration)
```
Comment (RaitoBezarius) [inline src/Branch-Off.md:21]: Wouldn't you change here to 23.05 ?

**Review by RaitoBezarius — COMMENTED** [review]
Shouldn't this PR change to 23.11 ?

File: src/Branch-Off.md (hunk 1 above)
Comment (dasJ) [reply src/Branch-Off.md:21]: Shouldn't matter, it's just an example to make it more clear imo

Comment (mweinelt) [reply src/Branch-Off.md:21]: Yeah, that's what I said. But it was apparently confusing to figsoda.

Comment (RaitoBezarius) [reply src/Branch-Off.md:21]: Yes, but then if it confuses someone once, and we are fixing this to unconfuse the next person, the example should move?

Comment (figsoda) [reply src/Branch-Off.md:21]: Yeah that would probably be better, hacking this thread to apply the suggestion
```suggestion
Example configuration: [nixos:release-23.05](https://hydra.nixos.org/jobset/nixos/release-23.05#tabs-configuration)
```

**Comment by figsoda** [comment]
One issue with 23.11 is https://hydra.nixos.org/jobset/nixos/release-23.11#tabs-configuration is not yet available, I think it should be fine as long as we are consistent with the first paragraph:

> For these steps "23.05" represents the current release tag and "23.11" represents the next
(6 months in the future) release tag. "22.11" is the last release which was released 6 months
ago.

**Comment by figsoda** [comment]
updated everything to 23.11, please lmk if I missed anything

File: src/Branch-Off.md [hunk 2]
```diff
@@ -132,7 +132,7 @@ Now we prepare the master branch for the next release after this one.
    ```bash
    git tag --annotate $NEWVER-pre
    git push upstream master $NEWVER-pre
-   git describe HEAD # should yield 23.05-pre
+   git describe HEAD # should yield 24.05-pre
```
Comment (wegank) [inline src/Branch-Off.md:135]: So `$NEWVER-pre` should actually be `24.05-pre` here?

**Pull request merged**